Fails to build on Raspberry Pi Pico2W

Hey, there appears to be an issue building for RP2350 based Pico2W.
This section doesn't seem correct, it prioritises the ARDUINO ifdef over the PICO_BUILD, which results in the error:
```
.pio\libdeps\pimoroni_pico_plus_2w\OneWireNg\src\platform\Platform_TimeCritical.cpp:19:1: error: 'tc_t' does not name a type
   19 | tc_t _tc[portNUM_PROCESSORS];
      | ^~~~
*** [.pio\build\pimoroni_pico_plus_2w\lib610\OneWireNg\platform\Platform_TimeCritical.cpp.o] Error 1
```
https://github.com/pstolarz/OneWireNg/blob/master/src/platform/Platform_TimeCritical.h#L104-L110

If I move the arduino/mbed ifdef checks after pico_build then the compilation completes as expected.

(I'm building against the pimoroni version of the rp2350 + RP2 wifi module as the [pico2w hasn't made it into the board definitions yet for MaxGerhardt's platformio repo](https://github.com/maxgerhardt/platform-raspberrypi/issues/86).)

--- a/src/platform/Platform_TimeCritical.h
+++ b/src/platform/Platform_TimeCritical.h
@@ -101,16 +101,10 @@ extern tc_t _tc[portNUM_PROCESSORS];
 #  define timeCriticalExit() \
     portCLEAR_INTERRUPT_MASK_FROM_ISR(_tc[xPortGetCoreID()].int_lev)
 # endif
-#elif defined(ARDUINO)
-# define timeCriticalEnter() noInterrupts()
-# define timeCriticalExit() interrupts()
-#elif defined(__MBED__)
-# define timeCriticalEnter() __disable_irq()
-# define timeCriticalExit() __enable_irq()
 #elif defined(PICO_BUILD)
 # include "hardware/sync.h"
 
-#define portNUM_PROCESSORS 2
+# define portNUM_PROCESSORS 2
 typedef struct {
     uint32_t int_lev;       /* saved interrupt level */
 } tc_t;
@@ -121,6 +115,12 @@ extern tc_t _tc[portNUM_PROCESSORS];
 
 # define timeCriticalExit() \
     restore_interrupts(_tc[get_core_num()].int_lev)
+#elif defined(ARDUINO)
+# define timeCriticalEnter() noInterrupts()
+# define timeCriticalExit() interrupts()
+#elif defined(__MBED__)
+# define timeCriticalEnter() __disable_irq()
+# define timeCriticalExit() __enable_irq()
 #else
 # ifndef OWNG_TEST
 #  warning "Time critical API unsupported for the target platform. Disabled."

